### PR TITLE
tests: unit tests for DoJSON journals

### PR DESCRIPTION
**Pull request by jacquerie**
Continues to address #1240.

- [x] Journals
    - [x] Test `issn`
    - [x] Test `coden`
    - [x] Test `publisher`
    - [x] Test `titles`
    - [x] Test `short_titles`
    - [x] Test `title_variants`
    - [x] ~~Test `name_variants`~~ (obsolete rule)

### CODE REVIEW

**Comment by jacquerie** [comment]
[This block](https://github.com/inspirehep/inspire-next/blob/1d3d2320ea11a7a947cda88a6283cead966f41ae/inspirehep/dojson/journals/fields/bd1xx.py#L57-L59) is not tested because it never happens on INSPIRE: https://inspirehep.net/search?ln=en&cc=Journals&p=022__b%3A%27hardcover%27&action_search=Search.

On the other hand, it sounded reasonable...

**Comment by jacquerie** [comment]
Ready for review.

File: inspirehep/dojson/journals/fields/bd1xx.py [hunk 1]
```diff
@@ -108,10 +108,3 @@ def title_variants(self, key, value):
     return {
         'title': value.get('a'),
     }
-
-
-@journals.over('name_variants', '^730..')
-@utils.for_each_value
-def name_variants(self, key, value):
-    """Variants of the name."""
-    return value.get('a')
```
Comment (kaplun) [inline inspirehep/dojson/journals/fields/bd1xx.py]: Is it normal that this disappeared?

**Comment by kaplun** [comment]
Why are you removing name variants? They are the core functionality of the Journal DB.

File: inspirehep/dojson/journals/fields/bd1xx.py (hunk 1 above)
Comment (jacquerie) [inline inspirehep/dojson/journals/fields/bd1xx.py]: `730__a` is already consumed by rule `title_variants`, which populates the field `title_variants`, which is in the schema, while `name_variants` isn't there: https://github.com/inspirehep/inspire-next/blob/303444d0e344d3a4c750d82c45c6094a6f224e05/inspirehep/modules/records/jsonschemas/records/journals.json

Comment (jacquerie) [inline inspirehep/dojson/journals/fields/bd1xx.py]: In any case, I'll add a bullet point to the commit message to say this.

File: inspirehep/modules/theme/templates/inspirehep_theme/format/record/Journal_HTML_detailed.tpl [hunk 2]
```diff
@@ -112,7 +112,7 @@
       </div>
       <div class="modal-body">
         <div style="text-align:left;">
-        {{ record['name_variants']|join(', ')}}
+        {{ record['title_variants']|join(', ')}}
```
Comment (jacquerie) [inline inspirehep/modules/theme/templates/inspirehep_theme/format/record/Journal_HTML_detailed.tpl]: In fact, their leftover use had to be amended.

**Pull request merged**